fdbcli/fdbcli.actor.cpp

```diff
@@ -2131,52 +2131,62 @@ ACTOR Future<bool> fileConfigure(Database db, std::string filePath, bool isNewDa
 
 // FIXME: Factor address parsing from coordinators, include, exclude
 
-ACTOR Future<bool> coordinators( Database db, std::vector<StringRef> tokens, bool isClusterTLS ) {
+ACTOR Future<bool> coordinators(Database db, std::vector<StringRef> tokens) {
 	state StringRef setName;
 	StringRef nameTokenBegin = LiteralStringRef("description=");
-	for(auto tok = tokens.begin()+1; tok != tokens.end(); ++tok)
+	for (auto tok = tokens.begin() + 1; tok != tokens.end(); ++tok) {
 		if (tok->startsWith(nameTokenBegin)) {
 			setName = tok->substr(nameTokenBegin.size());
 			std::copy( tok+1, tokens.end(), tok );
 			tokens.resize( tokens.size()-1 );
 			break;
 		}
-
-	bool automatic = tokens.size() == 2 && tokens[1] == LiteralStringRef("auto");
-
+	}
 	state Reference<IQuorumChange> change;
+	state bool force = false;
 	if (tokens.size()==1 && setName.size()) {
 		change = noQuorumChange();
-	} else if (automatic) {
-		// Automatic quorum change
-		change = autoQuorumChange();
 	} else {
 		state std::set<NetworkAddress> addresses;
 		state std::vector<StringRef>::iterator t;
-		for(t = tokens.begin()+1; t != tokens.end(); ++t) {
-			try {
-				// SOMEDAY: Check for keywords
-				auto const& addr = NetworkAddress::parse( t->toString() );
-				if (addresses.count(addr)){
-					printf("ERROR: passed redundant coordinators: `%s'\n", addr.toString().c_str());
-					return true;
-				}
-				addresses.insert(addr);
-			} catch (Error& e) {
-				if (e.code() == error_code_connection_string_invalid) {
-					printf("ERROR: '%s' is not a valid network endpoint address\n", t->toString().c_str());
-					return true;
+		state bool automatic = false;
+		for (t = tokens.begin() + 1; t != tokens.end(); ++t) {
+			if (*t == LiteralStringRef("auto")) {
+				// If auto, the other arguments are not relevant
+				// FORCE will not matter because auto will not choose
+				// unavailable processes as coordinators
+				automatic = true;
+				break;
+			} else if (*t == LiteralStringRef("FORCE")) {
```

Comment:
There's a doc string towards the top of this file that describes usage, and you should make it say `<auto | [FORCE] <address>+>`  (or something like that, but <> is required and [] is optional)